ext/datasource/consul/consul.go

```diff
@@ -0,0 +1,161 @@
+package consul
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/alibaba/sentinel-golang/ext/datasource"
+	"github.com/alibaba/sentinel-golang/logging"
+	"github.com/alibaba/sentinel-golang/util"
+	"github.com/cenkalti/backoff"
+	"github.com/hashicorp/consul/api"
+)
+
+type consulDataSource struct {
+	datasource.Base
+	*options
+
+	propertyKey string
+	kvQuerier   KVQuerier
+
+	quitCh        chan struct{}
+	isInitialized util.AtomicBool
+	waitIndex     uint64
+}
+
+var (
+	ErrNilConsulClient = errors.New("nil consul client")
+
+	logger = logging.GetDefaultLogger()
+)
+
+// NewConsulDataSource returns new consul datasource instance
+func NewConsulDataSource(propertyKey string, opts ...Option) (datasource.DataSource, error) {
+	var options = evaluateOptions(opts)
+	// if not consul client is specified, initialize from the configuration
+	if options.consulClient == nil {
+		client, err := api.NewClient(options.consulConfig)
```

Comment:
It mights cause nil pointer if options.consulConfig don't be assigned.
Besides, Is it better to force the user to pass in the client parameter manually?